FieldOpt/Reservoir/grid/eclgrid.cpp

```diff
@@ -21,178 +21,203 @@
-  }
-  else{
-    throw runtime_error("ECLGrid::GetCell(int i, int j, int k): Grid "
-                            "source must be defined before getting a cell.");
-  }
+    if (!IndexIsInsideGrid(i, j, k))
+        throw std::runtime_error("Error getting grid cell. Index (i, j, k) is outside grid.");
+
+    if (type_ == GridSourceType::ECLIPSE) {
+        int global_index = ecl_grid_reader_->ConvertIJKToGlobalIndex(i, j, k);
+        return GetCell(global_index);
+    }
+    else throw std::runtime_error("ECLGrid::GetCell: Grid source must be defined before getting a cell.");
 }
 
 Cell ECLGrid::GetCell(IJKCoordinate *ijk)
 {
-  // Check if IJK cell is inside overall (i.e., active+inactive) grid
-  if (!IndexIsInsideGrid(ijk)) {
-    throw runtime_error("ECLGrid::GetCell(*ijk): Error getting grid "
-                            "cell. Index (i, j, k) is outside grid.");
-  }
-
-  if (type_ == GridSourceType::ECLIPSE) {
-    int global_index = ecl_grid_reader_->ConvertIJKToGlobalIndex(ijk->i(), ijk->j(), ijk->k());
-    return GetCell(global_index);
-  }
-  else{
-    throw runtime_error("ECLGrid::GetCell(*ijk): Grid source must "
-                            "be defined before getting a cell.");
-  }
+    if (!IndexIsInsideGrid(ijk)) throw std::runtime_error("ECLGrid::GetCell: Index ijk is outside grid.");
+    if (type_ == GridSourceType::ECLIPSE) {
+        int global_index = ecl_grid_reader_->ConvertIJKToGlobalIndex(ijk->i(), ijk->j(), ijk->k());
+        return GetCell(global_index);
+    }
+    else throw std::runtime_error("ECLGrid::GetCell: Grid source must be defined before getting a cell.");
+}
+
+std::vector<int> ECLGrid::GetBoundingBoxCellIndices(double xi, double yi, double zi, double xf, double yf, double zf)
+{
+    double x_i, y_i, z_i, x_f, y_f, z_f;
+    x_i = std::min(xi, xf);
+    x_f = std::max(xi, xf);
+    y_i = std::min(yi, yf);
+    y_f = std::max(yi, yf);
+    z_i = std::min(zi, zf);
+    z_f = std::max(zi, zf);
+
+    int total_cells = Dimensions().nx * Dimensions().ny * Dimensions().nz;
+
+    std::vector<int> indices_list;
+    for (int ii = 0; ii < total_cells; ii++)
+    {
+        Cell cell;
+        // Try is here because we only want to get the list of active cells - that means defined cells
+        try
+        {
+            cell = GetCell(ii);
+            // Calculate cell size
+            double dx = (cell.corners()[5] - cell.corners()[4]).norm();
+            double dy = (cell.corners()[6] - cell.corners()[4]).norm();
+            double dz = (cell.corners()[0] - cell.corners()[4]).norm();
+            // std::cout << cell.center().x() << " : " << cell.center().y() << " : " << cell.center().z() << std::endl;
+            if ((cell.center().x() >= x_i - dx/1.7) && (cell.center().x() <= x_f + dx/1.7) &&
+                (cell.center().y() >= y_i - dy/1.7) && (cell.center().y() <= y_f + dy/1.7) &&
+                (cell.center().z() >= z_i - dz/1.7) && (cell.center().z() <= z_f + dz/1.7))
+            {
```

Comment:
The algorithm checks. I assume the .3 off is a slack to be sure the given cell is within the bounding box, and avoid numerical efforts. Though I think it is a little bit large for this purpose... on the other hand, I would not necessarily regarding the dx, dy, dz values as very accurate because dx_top may be different than dx_bottom, for example; maybe there is an ERT function that can provide mean values?